lib/ecto/migration/runner.ex

```diff
@@ -56,14 +58,31 @@ defmodule Ecto.Migration.Runner do
   end
 
   @doc """
-  Executes command tuples or strings.
+  Queued migration commands are executed.
+
+  Reverses the order commands are executed when doing a rollback
+  on a change/0 function.
+  """
+  def flush do
```

Comment:
You need to reset the commands in the agent. Otherwise flushing twice will execute the same commands twice! Try to do just a simple `get_and_update` call to the agent that retrieves the commands, direction and resets the commands in one pass!